### PR TITLE
Change block manager to accept a ArrayBuffer 

### PR DESCRIPTION
**Pull request by shivaram**
Attempt to fix #269. Existing tests pass, but I haven't added a new test to make sure things are fixed. This is partly because I was not sure if I should extend CacheTrackerSuite or add something to the Block Manager test cases.

### CODE REVIEW

File: core/src/main/scala/spark/storage/BlockManager.scala [hunk 1]
```diff
@@ -529,11 +532,18 @@ class BlockManager(val master: BlockManagerMaster, val serializer: Serializer, m
     }
   }
 
+  def put(blockId: String, values: Iterator[Any], level: StorageLevel, tellMaster: Boolean)
```
Comment (mateiz) [inline core/src/main/scala/spark/storage/BlockManager.scala:535]: Why didn't you make tellMaster have a default value there? Did the compiler complain due to the other put?

Comment (shivaram) [inline core/src/main/scala/spark/storage/BlockManager.scala:535]: Yes - The compiler gives an error that multiple overloaded methods cannot have a default value.

**Comment by mateiz** [comment]
This looks good to me. I would add a test in DistributedSuite actually, to just launch a cluster where the memory limit per worker is low (e.g. set spark.storage.memoryFraction = 0.01) and try caching and using some data on it.

File: core/src/main/scala/spark/storage/BlockManager.scala (hunk 1 above)
Comment (mateiz) [inline core/src/main/scala/spark/storage/BlockManager.scala:535]: OK, makes sense. Just add the test as I mentioned and it should be good to go.

**Comment by shivaram** [comment]
Added the test with the assumption that 512MB of memory times 0.0001 is around 50KB and the array should be larger than that.

**Comment by mateiz** [comment]
Thanks. It's a good bet because we do set the JVMs to exactly 512 MB of memory. I might also add a test where half the data fits and half doesn't.

**Pull request merged**